application/src/main/java/org/togetherjava/tjbot/commands/tag/TagManageCommand.java

```
@@ -0,0 +1,231 @@
+            case "deletetag" -> deleteTag(event, member);
+            case "createtag" -> createTag(event);
+            case "createidtag" -> createIdTag(event);
+            default -> throw new IllegalStateException();
+        }
+    }
+
+    private void editTag(SlashCommandEvent event) {
+        String tagId = event.getOption("id").getAsString();
+        String content = event.getOption(CONTENT).getAsString();
+
+        if (!tagSystem.exists(tagId)) {
+            event.reply(THIS_TAG_DOES_NOT_EXIST).setEphemeral(true).queue();
+
+            return;
+        }
+
+        tagSystem.put(tagId, content);
+
+        event
+            .replyEmbeds(new EmbedBuilder().setColor(Color.GREEN)
+                .setTitle(SUCCESS)
+                .setTimestamp(LocalDateTime.now())
+                .setFooter(event.getUser().getAsTag())
+                .setDescription("Successfully edited tag '" + tagId + "'")
+                .build())
+            .queue();
+    }
+
+    private void editIdTag(SlashCommandEvent event) {
+        String tagId = event.getOption(TAG_MINUS_ID).getAsString();
+        long messageId = event.getOption(MESSAGE_MINUS_ID).getAsLong();
+
+        if (!tagSystem.exists(tagId)) {
+            event.reply(THIS_TAG_DOES_NOT_EXIST).setEphemeral(true).queue();
+
+            return;
+        }
+
+        event.getMessageChannel().retrieveMessageById(messageId).queue(message -> {
+            tagSystem.put(tagId, message.getContentRaw());
+
+            event
+                .replyEmbeds(new EmbedBuilder().setColor(Color.GREEN)
+                    .setTitle(SUCCESS)
+                    .setTimestamp(LocalDateTime.now())
+                    .setFooter(event.getUser().getAsTag())
+                    .setDescription("Successfully edited tag '" + tagId + "'")
+                    .build())
+                .queue();
+        }, failure -> {
+            if (failure instanceof ErrorResponseException ex
```

Does this silently swallow other error conditions?

---

minor but `e` is preferred over `ex` (nice usage of pattern matching btw)